### PR TITLE
Include subfolders and print final summary

### LINKED ISSUE #9: Create summary message for roads testing
As a user I want to have a summary of test messaging. 

Acceptance Criteria:
- Summary of passed tests
- Summary of failed tests
- Summary of python errors (if not expected failures)

### LINKED ISSUE #10: Discover tests in subfolders of the configured test folder
Also include tests in subfolders of the configured test folder when creating the test menu.

### PR DESCRIPTION
**Pull request by dwsilk**
Closes #9
Closes #10

- Subfolders are now included in test discovery
- A final summary of the test results is printed to the QGIS Python Console after all testing
- Uses `unittest.TestLoader().discover()` instead of custom function to discover tests

### CODE REVIEW

**Comment by dwsilk** [comment]
Ready for review but not ready to merge, still a couple of super minor TODOs.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,24 @@ Change Log
...
+
+Added
+-----
+
+ * A final summary of the test results is printed to the QGIS Python Console after all testing
+
+Changed
+-------
+
+ * Subfolders are now included in test discovery
```
[the remark this answers is not in the document]
Comment (dwsilk) [reply CHANGELOG.rst]: This was one of my minor TODOs - good to confirm that you get it too. Seems a bit random as to when it appears and when it doesn't to me, but your hunch could be right. Will look into it more on Friday.

File: scriptassistant/plugin.py [hunk 2]
```diff
@@ -322,6 +358,69 @@ def prepare_test(self, test_name):
...
+            print "-" * len(f[0].__str__())
+            print f[1]
+        successes = self.aggregated_test_result.testsRun - (
+            len(self.aggregated_test_result.errors) +
+            len(self.aggregated_test_result.failures) +
+            len(self.aggregated_test_result.expectedFailures) +
+            len(self.aggregated_test_result.unexpectedSuccesses) +
+            len(self.aggregated_test_result.skipped)
+        )
+        print """
```
[the remark this answers is not in the document]
Comment (dwsilk) [reply scriptassistant/plugin.py]: Oops, yep. Another option I had thought about earlier is only printing rows that have a value > 0 ..?

[the remark this answers is not in the document]
Comment (pernlofgren) [reply scriptassistant/plugin.py]: thats a good idea i reckon

File: CHANGELOG.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dwsilk) [reply CHANGELOG.rst]: Okay this only occurs for me when the /test/ folder is inside the /roads/ plugin folder. Once it is moved outside /roads/ I don't get it anymore...

File: scriptassistant/plugin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dwsilk) [reply scriptassistant/plugin.py]: Done

**Comment by pkinglinz** [comment]
I"m thinking we don't, if it was a single small print out absolutely, but since it will be printing a number of results I would keep it minimal with only the valid results so the user doesn't have to scroll through junk

**Pull request merged**